.github/workflows/e2e.yml

```diff
@@ -51,7 +39,20 @@ jobs:
       run: docker run -d -p 8080:8080 --name haystack_test_weaviate --env AUTHENTICATION_ANONYMOUS_ACCESS_ENABLED='true' --env PERSISTENCE_DATA_PATH='/var/lib/weaviate' --env ENABLE_EXPERIMENTAL_BM25='true' --env DISK_USE_READONLY_PERCENTAGE='95' semitechnologies/weaviate:1.17.2
 
     - name: Install Haystack
-      run: pip install .
+      run: pip install .[inference,elasticsearch7,faiss,weaviate,opensearch,dev]
+
+    - name: Cache HF models
+      id: cache-hf-models
+      uses: actions/cache@v3
+      with:
+        path: ~/.cache/huggingface/transformers/
+        key: hf-models
```

Comment:
A fixed `key` for caching is a really bad idea, this will never invalidate it. 

We should generate an hash given the name of the models. You can see an example of how to generate an hash given a file in [actions/cache documentation](https://github.com/actions/cache#creating-a-cache-key).

I would save the models below in a txt file somewhere and use that to generate the hash, then in the step below I'd load the models from the files.

Something like this:
```
      - name: Download models
        if: steps.cache-hf-models.outputs.cache-hit != 'true'
        shell: python
        run: |
          from transformers import AutoModel
          with open("./e2e/models_to_cache.txt") as file:
            AutoModel.from_pretrained(file.readline())
```

Be careful with the path as it must be relative to the root of the project.